Libraries/dotNetRDF/Query/Builder/QueryBuilder.cs

```diff
@@ -377,10 +384,7 @@ private void BuildRootGraphPattern(SparqlQuery query)
         {
             var rootGraphPattern = RootGraphPatternBuilder.BuildGraphPattern(Prefixes);
 
-            if (!rootGraphPattern.IsEmpty)
-            {
-                query.RootGraphPattern = rootGraphPattern;
-            }
+            query.RootGraphPattern = rootGraphPattern;
```

Comment:
This one change seems unrelated but I found that `VALUES` to not switch `IsEmpty`. 

Does is make any difference if the root in null or not?

Reply:
I think if the root graph pattern is null / empty then you should just end up with the values in the result set. Not 100% sure, but I don't think this is a problem ;-)